src/sentry/models/groupopenperiod.py

```diff
@@ -0,0 +1,43 @@
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, Model, region_silo_model, sane_repr
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+@region_silo_model
+class GroupOpenPeriod(Model):
+    """
+    A GroupOpenPeriod is a period of time where a group is considered "open",
+    i.e. having a status that is not resolved. This is primarily used for
+    detector-based issues to track the period of time that an issue is open for.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    project = FlexibleForeignKey("sentry.Project")
+    group = FlexibleForeignKey("sentry.Group")
+    resolution_activity = FlexibleForeignKey(
+        "sentry.Activity", null=True, on_delete=models.SET_NULL
+    )
+
+    # if the user is not set, it's assumed to be the system
+    user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete="SET_NULL")
+    date_started = models.DateTimeField(default=timezone.now)
+    date_ended = models.DateTimeField(null=True)
```

Comment:
Do we want to guard against overlapping open periods?

Reply:
TIL about `ExclusionConstraint` - not quite sure if I've set it up right